python_modules/dagster/dagster/_core/snap/job_snapshot.py

```diff
@@ -136,7 +133,7 @@ def __new__(
         graph_def_name: str,
         metadata: Optional[Mapping[str, RawMetadataValue]],
     ):
-        return super(JobSnapshot, cls).__new__(
+        return super().__new__(
```

Comment:
you can remove all these check calls in this constructor since the generated one will handle them (more efficiently)